Simulator.cpp

```diff
@@ -6,104 +6,1028 @@
-            Mbas0 += cur->volume*elem->sumPhi/(2*M_PI)*cur->q_m;
-            MbasDelta += cur->volume*elem->sumPhi/(2*M_PI)*sim->thermoneutral*deltaQMetabolic(cur->q_m,T[Tidx],Tn[Tidx],sim->Q10);
-            Tidx++;
-            for(int secIdx = 0; secIdx<elem->nSectors; ++secIdx){
-                SECTOR::Sector* sect = elem->sectors[secIdx];
-                for(int washIdx = 1; washIdx<elem->nWashers; ++washIdx){
+            Mbas0 += body->V[idx]*cur->q_m;
+            MbasDelta += body->V[idx]*sim->thermovariant*sim->transient*deltaQMetabolic(cur->q_m,T[idx],T0[idx]);
+            idx++;
+            for(sectIdx = 0; sectIdx<elem->nSectors; ++sectIdx){
+                SECTOR::Sector* sect = elem->sectors[sectIdx];
+                for(washIdx = 1; washIdx<elem->nWashers; ++washIdx){
                     cur = elem->washers[washIdx];
-                    Mbas0 += cur->volume*sect->phi/(2*M_PI)*cur->q_m;
-                    MbasDelta += cur->volume*sect->phi/(2*M_PI)*sim->thermoneutral*deltaQMetabolic(cur->q_m,T[Tidx],Tn[Tidx],sim->Q10);
-                    Tidx++;
+                    Mbas0 += body->V[idx]*cur->q_m;
+                    MbasDelta += body->V[idx]*sim->thermovariant*sim->transient*deltaQMetabolic(cur->q_m,T[idx],T0[idx]);
+                    idx++;
                 }
             }
         }
+        assert(!isnan(abs(Sh)));
+        assert(!isnan(abs(MbasDelta)));
+        Mbas0 = 58.2;
         double act = 1 + (Sh+MbasDelta)/Mbas0*MperWork; // MET
         double etaW;
         if(act < 1.6)
             etaW = 0.05; // W/W
         else
             etaW = 0.2*tanh(body->b1*act+body->b0); // W/W
-        M = act*Mbas0/actBas; // W
+        assert(!isnan(abs(act)));
+        M = act*Mbas0/body->actBas; // W
+        assert(!isnan(abs(M)));
         H = M*(1-etaW)-Mbas0; // W
-        Qresp = computeQresp(); // W
-
+        assert(!isnan(abs(H)));
+        QResp = computeQresp(); // W
+        assert(!isnan(abs(QResp)));
     }
-    double computeQresp(){
+    double SimulationInstance::computeQresp(){
+        assert(!isnan(abs(sim->Pair)));
+        assert(!isnan(abs(sim->Tair)));
         return 3.45*M*(0.028+6.5e-5*sim->Tair-4.98e-6*sim->Pair)+
             1.44e-3*M*(32.6-0.934*sim->Tair+1.99e-4*sim->Pair);
     }
-    double deltaQMetabolic( double q, double T, double Tn, double Q10 )
+    double SimulationInstance::deltaQMetabolic( double q, double T, double Tn)
     {
-        return q*(pow(Q10,(T-Tn)/10.0)-1);
+        return q*(pow(sim->Q10,(T-Tn)*sim->thermovariant*sim->transient/10.0)-1);
+    }
+    double SimulationInstance::project( double cur, double prv ){
+        assert(!isnan(abs(cur)));
+        if(sim->transient ==0 || sim->thermovariant==0){
+            return cur;
+        }
+        else{
+            assert(!isnan(abs(prv)));
+            return cur+(cur-prv);
+        }
+        
+    }
+    void SimulationInstance::projectBodyValues(){
+        // MNxt = project(M,MPrv);
```

Comment:
Delete commented lines. Does this function not do anything? If not, can we remove it?

Reply:
This I want some input on. Structure-wise it makes sense, there should be a function to project body-wide values just like there is for nodes and elements. But, there aren't any using the Konstas case that needed it. I'd delete the comments but leave the function.

Reply:
Ah okay, in that case I'd say keep it and make the comment that the Konstas case does not need to project the body-wide values in the function body